src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs

```diff
@@ -0,0 +1,339 @@
+            _registryManager.DeleteValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonPassword);
+            
+            SetValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonCount, "");
+            _registryManager.DeleteValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonCount);
+
+            SetValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogon, "1");
+
+            SetValue(hive, RegistryConstants.SubKeys.ShutdownReasonDomainPolicy, RegistryConstants.ValueNames.ShutdownReason, "0");
+            SetValue(hive, RegistryConstants.SubKeys.ShutdownReasonDomainPolicy, RegistryConstants.ValueNames.ShutdownReasonUI, "0");
+
+            SetValue(hive, RegistryConstants.SubKeys.LegalNotice, RegistryConstants.ValueNames.LegalNoticeCaption, "");
+            SetValue(hive, RegistryConstants.SubKeys.LegalNotice, RegistryConstants.ValueNames.LegalNoticeText, "");
+        }
+
+        private void UpdateUserSpecificRegistrySettings(string domainName, string userName)
+        {
+            if (!_windowsServiceHelper.HasActiveSession(domainName, userName))
+            {
+                string securityId = _windowsServiceHelper.GetSecurityId(domainName, userName);
+                UpdateRegistrySettingsForDifferentUser(securityId);                
+            }
+            else
+            {
+                UpdateRegistrySettingsForCurrentUser();
+            }
+        }
+
+        private void UpdateRegistrySettingsForCurrentUser()
+        {
+            var hive = RegistryHive.CurrentUser;
+            SetValue(hive, RegistryConstants.SubKeys.ScreenSaver, RegistryConstants.ValueNames.ScreenSaver, "0");
+            SetValue(hive, RegistryConstants.SubKeys.ScreenSaverDomainPolicy, RegistryConstants.ValueNames.ScreenSaver, "0");
+            SetValue(hive, RegistryConstants.SubKeys.StartupProcess, RegistryConstants.ValueNames.StartupProcess, GetStartupCommand());
+        }
+
+        private void UpdateRegistrySettingsForDifferentUser(string securityId)
+        {
+            var hive = RegistryHive.Users;
+            SetValue(hive, $"{securityId}\\{RegistryConstants.SubKeys.ScreenSaver}", RegistryConstants.ValueNames.ScreenSaver, "0");
+            SetValue(hive, $"{securityId}\\{RegistryConstants.SubKeys.ScreenSaverDomainPolicy}", RegistryConstants.ValueNames.ScreenSaver, "0");
+            SetValue(hive, $"{securityId}\\{RegistryConstants.SubKeys.StartupProcess}", RegistryConstants.ValueNames.StartupProcess, GetStartupCommand());
+        }
+
+        private string GetStartupCommand()
+        {
+            //startup process            
+            string cmdExePath = System.Environment.GetEnvironmentVariable("comspec");
+            if (string.IsNullOrEmpty(cmdExePath))
+            {
+                cmdExePath = "cmd.exe";
```

Comment:
agree that it can happen for 'start' but cmd.exe is accessible without the path as well. Am I missing something?

---
In reply to: [118597790](https://github.com/Microsoft/vsts-agent/pull/985#discussion_r118597790) [](ancestors = 118597790)